nomad/resources_endpoint.go

```diff
@@ -0,0 +1,138 @@
+package nomad
+
+import (
+	"fmt"
+	"github.com/hashicorp/go-memdb"
+	"github.com/hashicorp/nomad/nomad/state"
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+// truncateLimit is the maximum number of matches that will be returned for a
+// prefix for a specific context
+const (
+	truncateLimit = 20
+)
+
+// allContexts are the available contexts which searched to find matches for a
+// given prefix
+var (
+	allContexts = []string{"allocs", "nodes", "jobs", "evals"}
+)
+
+// Resource endpoint is used to lookup matches for a given prefix and context
+type Resources struct {
+	srv *Server
+}
+
+// getMatches extracts matches for an iterator, and returns a list of ids for
+// these matches.
+func getMatches(iter memdb.ResultIterator) ([]string, bool) {
+	var matches []string
+	isTruncated := false
+
+	for i := 0; i < truncateLimit; i++ {
+		raw := iter.Next()
+		if raw == nil {
+			break
+		}
+
+		var id string
+		switch raw.(type) {
+		case *structs.Job:
+			id = raw.(*structs.Job).ID
+		case *structs.Evaluation:
+			id = raw.(*structs.Evaluation).ID
+		case *structs.Allocation:
+			id = raw.(*structs.Allocation).ID
+		case *structs.Node:
+			id = raw.(*structs.Node).ID
+		default:
+			//s.logger.Printf("[ERR] nomad: unexpected type for resources context; not a job, allocation, node, or evaluation")
```

Comment:
You can either make it a method of `Resources` or have it return an error. Also we should just use the variable so that if contexts get added it will automatically propagate.